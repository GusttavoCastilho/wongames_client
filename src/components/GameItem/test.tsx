import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import GameItem from '.'

const props = {
  img: 'https://source.unsplash.com/user/willianjusten/151x70',
  title: 'Red Dead Redemption 2',
  price: 'R$ 215,00'
}

describe('<GameItem />', () => {
  it('should render the item', () => {
    renderWithTheme(<GameItem {...props} />)

    expect(
      screen.getByRole('img', { name: /red dead redemption 2/i })
    ).toHaveAttribute('src', props.img)

    expect(
      screen.getByRole('heading', { name: /red dead redemption 2/i })
    ).toBeInTheDocument()

    expect(screen.getByText('R$ 215,00')).toBeInTheDocument()
  })

  it('should render the icon with download link', () => {
    const downloadLink = 'https://link'

    renderWithTheme(<GameItem {...props} downloadLink={downloadLink} />)

    expect(
      screen.getByRole('link', { name: `Get ${props.title} here` })
    ).toHaveAttribute('href', downloadLink)
  })
})
