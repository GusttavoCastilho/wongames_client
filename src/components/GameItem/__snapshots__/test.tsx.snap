// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameItem /> should render the item 1`] = `
.c0 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c2 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c4 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c5 {
  color: #FAFAFA;
  padding: 0.2rem 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c2 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt="Red Dead Redemption 2"
        src="https://source.unsplash.com/user/willianjusten/151x70"
      />
    </div>
    <div
      class="c3"
    >
      <h3
        class="c4"
      >
        Red Dead Redemption 2
      </h3>
      <div
        class="c5"
      >
        R$ 215,00
      </div>
    </div>
  </div>
</div>
`;
